### ISSUE TITLE
"Waterfox is already running, but isn't responding."

### ISSUE THREAD

**Issue by FloridaMan7588** [flathub/net.waterfox.waterfox#5]
When opening a URL from another application using xdg-open, an error occurs stating

 "Waterfox is already running, but is not responding. To use Waterfox, you must first close the existing Waterfox process, restart your device, or use a different profile."

When running xdg-open from the command line, the error 
"JavaScript error: resource://gre/modules/XULStore.jsm, line 66: Error: Can't find profile directory."
is thrown. 

I believe this has something to do with the browser expecting the '.waterfox' directory to be in your home folder. I will look for a remedy through configuration, but as profile names are random ASCII spam, this may only be resolved by changing from using persist to filesystem in the finish args. This is what upstream is doing with the official Firefox package.

**Comment by Danetterry** [comment]
I changed permission filesystem to home. Now links will open, but in a new user profile.

**Comment by soko7awen** [comment]
When given full session bus access the error does not occur. I am currently attempting to find out which policy rule would be able to resolve the issue specifically.

**Comment by FloridaMan7588** [comment]
Interesting, I thought this was a filesystem issue at first but the bus access issue would be interesting to look into. I'll do some testing as well.

**Comment by FloridaMan7588** [comment]
I did some digging with [bustle](https://flathub.org/apps/org.freedesktop.Bustle) (Very useful program), and found that the problem is caused by an upstream issue, although it could possibly be fixed with a policy rule. It was caused by the application using a different name than what I assumed on the session bus. At build time it was set to use the session name of `org.mozilla.waterfox` as opposed to the `net.waterfox.waterfox` name that follows the reverse DNS formatting. I will submit a PR with the own-name changed in the manifest, but hopefully this can be resolved with a policy or preference change in `about:config` or similar. From the looks of it, it isn't in the config but it may be fixed in another way. Until then, I will change the manifest directly.
